MIA.cpp

```diff
@@ -70,13 +71,25 @@ try
 		std::make_shared<mia::modules::SerializationModule>()
 		});
 
-	for (const auto& x : dynamicLibs)
+	for (std::size_t x = 0; x < dynamicLibs.size(); ++x)
 	{
-		const auto miaModule = mia::GeneratorModule::loadFromLibrary(*x);
+		const auto miaModule = mia::GeneratorModule::loadFromLibrary(*dynamicLibs[x]);
 		if (miaModule)
 		{
 			app.registerModule(miaModule);
 		}
+		else
+		{
```

Comment:
This should be handled in GeneratorModule, pass module name there or store it in the module instead